### ISSUE TITLE
[Feature Request] Open web image and non-popup progress indicator

### ISSUE THREAD

**Issue by vguttmann** [mitxela/plotterfun#15]
I would wish that I could directly open images from a URL, and that, if plotterfun is recalculating, the popup wouldn't shift the whole layout when recalculating. 

I am using Firefox 85.0.1 64-bit on Ubuntu 20.04.1 LTS.

**Comment by mitxela** [comment]
Open from URL could be nice.

By popup do you mean the message at the bottom that says things like "optimizing route"? Because that's `position:fixed` and shouldn't be shifting anything. Could you tell me which algorithm you're using, and how big the source image is, and how big your browser window is? Or better yet, record a quick video of the problem.

**Comment by vguttmann** [comment]
No, I mean this spinny thing (like the windows 7 "busy" cursor), that appears above all the sliders whenever I change a value. My laptop is not the fastest, so you may haven't even seen that. Also, linedraw freezes, but that could also be down to my slow machine.

**Comment by vguttmann** [comment]
Edit: Linedraw happens to freeze with those two images:
![image0](https://user-images.githubusercontent.com/61903989/107974871-82421800-6fb7-11eb-9e76-9752ed4dcda5.jpg)
![image1](https://user-images.githubusercontent.com/61903989/107974880-853d0880-6fb7-11eb-9629-79651ad64956.jpg)

This image, which was made in exactly the same way (converting to grayscale in GIMP and adjusting the curve for better contrast), works completely fine:
![39998eb634f3ce3d5107f053b369f19e1](https://user-images.githubusercontent.com/61903989/107974990-af8ec600-6fb7-11eb-8da5-0b68d60a9abd.png)

**Comment by vguttmann** [comment]
Sooo, I checked the site, and the animation is called "loading.gif (https://mitxela.com/plotterfun/loading.gif), and is in the algorithm selection box. It has the image id "buffering", and appears every time I change a parameter.

**Comment by mitxela** [comment]
Something is quite wrong if the buffer wheel image is appearing whenever you change a parameter. It's only supposed to be needed when it's loading something over the network. As soon as an algorithm is loaded, it gets cached. Are you running this locally or is it hosted?

As for linedraw it will take a very long time to run on images which are big with lots of fine detail like that, it will try and trace the contours of every shading stroke. If you blur the image a bit first it will work better.

**Comment by vguttmann** [comment]
I am using plotterfun as hosted on your website. I checked with my smartphone, using both firefox and chrome (chrome doesn't belong on my PC!), and the animation appeared every time, however, it always was in line with the algorithm dropdown, no matter if I viewed the mobile site or the desktop site (if that actually matters with your site).

And yes, blurring the image worked. I guess that was to be expected with a pencil-drawn image.

**Comment by mitxela** [comment]
How's that? shouldn't move the layout when it's shown now, anyway.

**Comment by vguttmann** [comment]
Works like a charm on all devices and browsers. Thank you!
